File: 1190-Reverse-Parentheses/1190.java
class Solution {
  public String reverseParentheses(String s) {
    Deque<String> stack = new LinkedList<String>();
    StringBuffer sb = new StringBuffer();
    for (int i = 0; i < s.length(); i++) {
      char ch = s.charAt(i);
      if (ch == '(') {
        stack.push(sb.toString());
        sb.setLength(0);
      } else if (ch == ')') {
        sb.reverse();
        sb.insert(0, stack.pop());
      } else {
        sb.append(ch);
      }
    }
    return sb.toString();
  }
}

File: 1101-1200/1190-Reverse-Parentheses/1190.go
func reverseParentheses(s string) string {
	stack := [][]byte{}
	str := []byte{}
	for i := range s {
		if s[i] == '(' {
			stack = append(stack, str)
			str = []byte{}
		} else if s[i] == ')' {
			for j, n := 0, len(str); j < n/2; j++ {
				str[j], str[n-1-j] = str[n-1-j], str[j]
			}
			str = append(stack[len(stack)-1], str...)
			stack = stack[:len(stack)-1]
		} else {
			str = append(str, s[i])
		}
	}
	return string(str)
}